sympy/geometry/parabola.py

```diff
@@ -407,9 +408,135 @@ def vertex(self):
+        ========
+
+        >>> from sympy import Parabola, Point, Line
+        >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+        >>> p1.centroid(Line(Point(0, 0), Point(2, 0)))
+        Point2D(0, 8/5)
+
+        """
+
+        if(self.directrix.slope == 0):
+            return Point(self.vertex[0], (2*self.vertex[1] + 3*l.p1[1])/5)
+
+        if(self.directrix.slope == S.Infinity):
+            return Point((2*self.vertex[0] + 3*l.p1[0])/5, self.vertex[1])
+
+    def second_moment_of_area(self, l, point=None):
+        """Returns the second moment and product moment of area of a Parabola bounded by line l.
+
+        Parameters
+        ==========
+
+        l : Line
+            l is the line perpendicular to symmetric axis
+            up to which second moment of area is to be calculated.
+        point : Point, two-tuple of sympifiable objects, or None(default=None)
+            point is the point about which second moment of area is to be found.
+            If "point=None" it will be calculated about the axis passing through the
+            centroid of the polygon.
+
+        Returns
+        =======
+
+        I_xx, I_yy, I_xy : number or sympy expression
+            I_xx, I_yy are second moment of area of a two dimensional polygon.
+            I_xy is product moment of area of a two dimensional polygon.
+
+        Examples
+        ========
+        >>> from sympy import Parabola, Point, Line
+        >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+        >>> p1.second_moment_of_area(Line(Point(0, 0), Point(2, 0)))
+        (94208/175, 8192/15, 0)
+
+        """
+        a = self.focal_length
+        I_xy_v = 0;
+        c = self.centroid(l)
+        Ar = self.area(l)
+        if(self.directrix.slope == 0):
+            I_xx_v = 32*sqrt((a**3))*sqrt(Abs((self.vertex[1] - l.p1[1])**5))/15
+            I_yy_v = 8*sqrt(a*(Abs((self.vertex[1] - l.p1[1]))**7))/7
+
+            I_xx_c = I_yy_v + Ar*((c[1] - self.vertex[1])**2)
+            I_yy_c = I_xx_v
+            I_xy_c = I_xy_v
+            if point is None:
+                return I_xx_c, I_yy_c, I_xy_c
+
+            I_xx = I_xx_c + Ar*((c[1] - point[1])**2)
+            I_yy = I_yy_c + Ar*((c[0] - point[0])**2)
+            I_xy = I_xy_c + Ar*((point[0]-c[0])*(point[1]-c[1]))
+
+            return I_xx, I_yy, I_xy
+
+        if(self.directrix.slope == S.Infinity):
+            I_xx_v = 32*sqrt((a**3))*sqrt((Abs((self.vertex[0] - l.p1[0]))**5))/15
+            I_yy_v = 8*sqrt(a*(Abs((self.vertex[0] - l.p1[0]))**7))/7
+
+            I_xx_c = I_xx_v
+            I_yy_c = I_yy_v - Ar*((c[0] - self.vertex[0])**2)
+            I_xy_c = I_xy_v
+            if point is None:
+                return I_xx_c, I_yy_c, I_xx_c
+
+            I_xx = I_xx_c + Ar*((c[1] - point[1])**2)
+            I_yy = I_yy_c + Ar*((c[0] - point[0])**2)
+            I_xy = I_xy_c + Ar*((point[0]-c[0])*(point[1]-c[1]))
+
+            return I_xx, I_yy, I_xy
```

Comment:
I'm not really a fan of adding this feature. A parabola is not a closed shape and thus doesn't have an area, a centroid, or second moment of area, etc. I think a better direction is to implement shapes that can be constructed by boolean operations of various geometric elements.

Reply:
ok, I will implement shapes that can be constructed by boolean operations of various geometric elements. It will use `integrals` module to find the area, centroid and second moment of area, etc.

should I implement something related to the plotting of shapes that will be constructed by boolean operations of the various geometric element using `Plotting` Module?